pkgs/checks/lib/src/extensions/async.dart

```diff
@@ -26,6 +26,29 @@ extension FutureChecks<T> on Check<Future<T>> {
     });
   }
 
+  /// Expectst that the `Future` never completes as a value or an error.
+  ///
+  /// Immediately returns and does not cause the test to remain running if it
+  /// ends.
+  /// If the future completes at any time, raises a test failure. This may
+  /// happen after the test has already appeared to succeed.
+  ///
+  /// Not compatible with [softCheck] or [softCheckAsync] since there is no
+  /// concrete end point where this condition has definitely succeeded. Will
+  /// never cause a soft check to fail.
+  void doesNotComplete() {
+    context.expectLate(() => ['does not complete as value or error'],
+        (actual, reject) {
+      unawaited(actual.then((r) {
+        reject(Rejection(actual: 'A future that completed to ${literal(r)}'));
+      }, onError: (e, st) {
+        reject(Rejection(
+            actual: 'A future that completed as an error:',
+            which: ['threw ${literal(e)}', ...st.toString().split('\n')]));
```

Comment:
We should probably use a LineSplitter here just to be a bit more defensive

Reply:
Done.